14/solve_1.rb

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env ruby
+fail 'Script needs two parametes: <input_file> <time_limit>' if ARGV.length != 2
+lines = File.readlines(ARGV[0]).map(&:strip)
+total_seconds = ARGV[1].to_i
+
+def count_distance(speed, time, rest_time, total_seconds)
+  distance = 0
+  while total_seconds >= 0
+    if time < total_seconds
```

Comment:
not complaint, just idea:
distance += speed * (time < total_seconds ? time : total_seconds)